magma/circuit.py

```diff
@@ -518,10 +519,23 @@ def __new__(metacls, name, bases, dct):
         self.default_kwargs = dct.get('default_kwargs', {})
         self.firrtl = None
 
+        # If enabled, we first set interface ports undriven so it is treats a
```

Comment:
I feel like these passes should happen at compile time (or some later time), not at define time (i.e. in the metaclass). I know putting it here means you avoid getting a bunch of `ERROR`'s but those ERROR's aren't blocking anyway, and it may be a good idea to still have those. I feel like it would be better to have another kwarg in m.compile that will add all of these, ala `drive_undriven` and `terminate_unused`. On that note I would change the names of the passes to this. "ignore" seems like passive suppression of warnings rather than actively changing things to avoid them. Also, we could have a single function (with those names) that would apply that pass on a definition, so the user could run that pass without dealing with the "pass" itself.

If we want to configure whether or not we `ERROR` on undriven things, I think we should add a separate flag/config for that -- that would just configure how we report those things, not modify anything.

Reply:
updated to use those kwargs and passes at the end and added a flag to ignore undriven errors